(feat) O3-2353: Use demo data content package

NethmiRodrigo opened the pull request:
This PR updates the reference application to [use the demo data content package](https://github.com/openmrs/openmrs-content-referenceapplication-demo). Here's what I did:

1. Removed the backend configurations that will get introduced by the demo data content package.
2. Removed the frontend configuration `config-core_demo.json`.
3. Updated the `SPA_CONFIG_URL` to not have `config-core_demo.json`.
4. Updated the frontend Dockerfile to not copy the `config-core_demo.json` file.
5. Updated the distro properties file to include the demo data content package.
6. @jayasanka-sack created a bamboo plan for the demo ref app content package - https://ci.openmrs.org/browse/CH-CONREFD

jayasanka-sack reviewed (changes requested):
Thanks @NethmiRodrigo! We can remove this block since we don’t need local config anymore.:
https://github.com/openmrs/openmrs-distro-referenceapplication/blob/7d3691cb8c91584c8b401143500f55fbab6805fd/distro/pom.xml#L279

jayasanka-sack commented on frontend/config-core_demo.json:
```diff
@@ -1,52 +1 @@
-    "Brand color #1": "#005d5d"
-  }
-}
+{}
```
new line

NethmiRodrigo commented:
> openmrs-distro-referenceapplication/distro/pom.xml

@jayasanka-sack sorry, which block?

NethmiRodrigo commented:
> Thanks @NethmiRodrigo! We can remove this block since we don’t have local config anymore.:
> 
> https://github.com/openmrs/openmrs-distro-referenceapplication/blob/7d3691cb8c91584c8b401143500f55fbab6805fd/distro/pom.xml#L279

@jayasanka-sack since we don't have the local config anymore, it would make sense to delete the whole `distro/configuration` folder right (and not have the empty configuration folder)?

jayasanka-sack commented:
@NethmiRodrigo  yep.

jayasanka-sack commented:
Anyway, empty directories are not retained in Git.

dkayiwa commented on frontend/config-core_demo.json:
```diff
@@ -1,52 +1 @@
-{
-  "@openmrs/esm-patient-chart-app": {
-    "restrictByVisitLocationTag": true,
-    "showUpcomingAppointments": true,
```
I guess you forgot to remove this from: https://github.com/openmrs/openmrs-distro-referenceapplication/blob/main/frontend/Dockerfile#L39

NethmiRodrigo replied:
Done!

jayasanka-sack replied:
The reason it fails when the above line + file is removed because that config file is no longer in the html directory of the container.

The frontend configs from content packages are loaded to `distro/target/sdk-distro/web/openmrs_spa` directory from the sdk. We should somehow copy all the frontend config files to the html directory of the frontend container.

dkayiwa commented on docker-compose.yml:
```diff
@@ -49,7 +49,7 @@ services:
     restart: "unless-stopped"
     command: "mysqld --character-set-server=utf8mb4 --collation-server=utf8mb4_general_ci"
     healthcheck:
-      test: "mysql --user=${OMRS_DB_USER:-openmrs} --password=${OMRS_DB_PASSWORD:-openmrs} --execute \"SHOW DATABASES;\""
```
Did you touch this accidentally? :)

NethmiRodrigo replied:
Prettier had done this on save

NethmiRodrigo commented:
Before I deleted the frontend configuration file `config-core_demo.json` and updated the docker files to not use it, the e2e tests for form builder, core and cohort builder passed. But now they're timing out

wikumChamith reviewed (commented):
@NethmiRodrigo I am getting an error when running `docker compose build`

`yaml: line 52: mapping values are not allowed in this context`

wikumChamith commented:
> @NethmiRodrigo I am getting an error when running `docker compose build`
> 
> `yaml: line 52: mapping values are not allowed in this context`

I am only getting this error with your latest changes.
https://github.com/openmrs/openmrs-distro-referenceapplication/pull/885/commits/20f778c64a706ab0a8755e982eaab29a9a579c8d

dkayiwa commented:
@NethmiRodrigo this creates the patient demo data. But does not put the `Login Location` tag to the locations. And as a result, the locations picker remains empty.

NethmiRodrigo commented:
> @NethmiRodrigo this creates the patient demo data. But does not put the `Login Location` tag to the locations. And as a result, the locations picker remains empty.

@dkayiwa sorry what do you mean?

wikumChamith commented:
> @NethmiRodrigo this creates the patient demo data. But does not put the `Login Location` tag to the locations. And as a result, the locations picker remains empty.

@dkayiwa I do not see that error locally when running with these changes.

dkayiwa commented:
@NethmiRodrigo let me ask it differently, is the demo data content package supposed to also contain the contents of this file? https://github.com/openmrs/openmrs-distro-referenceapplication/blob/main/distro/configuration/locations/locations-core_demo.csv

@wikumChamith how many locations do you have in your location picker list?

wikumChamith commented:
> @NethmiRodrigo let me ask it differently, is the demo data content package supposed to also contain the contents of this file? https://github.com/openmrs/openmrs-distro-referenceapplication/blob/main/distro/configuration/locations/locations-core_demo.csv
> 
> @wikumChamith how many locations do you have in your location picker list?

57

![Uploading Screenshot from 2025-02-05 18-48-18.png…]()

NethmiRodrigo commented:
@dkayiwa yes it does - https://github.com/openmrs/openmrs-content-referenceapplication-demo/blob/main/configuration/backend_configuration/locations/locations-core_demo.csv

dkayiwa commented:
Then in that case this pull request Looks good to me. 😊

jayasanka-sack commented:
No wait.., this introduces more test failures. Earlier, it was only the chart and management that were failing, which was because of an error that had nothing to do with content packages.

NethmiRodrigo commented:
> No wait.., this introduces more test failures. Earlier, it was only the chart and management that were failing, which was because of an error that had nothing to do with content packages.
@jayasanka-sack I agree... I made a PR to revert this

vasharma05 commented:
> No wait.., this introduces more test failures. Earlier, it was only the chart and management that were failing, which was because of an error that had nothing to do with content packages.

What's the exact issue if someone can help me understand the same.
Thanks!

jayasanka-sack commented:
> > No wait.., this introduces more test failures. Earlier, it was only the chart and management that were failing, which was because of an error that had nothing to do with content packages.
> 
> What's the exact issue if someone can help me understand the same. Thanks!

@vasharma05 there were several issues:

1) 7d3691cb8c91584c8b401143500f55fbab6805fd introduced a bug that has to do with global variables
2) Then this PR is opened. Changes here are not related to the above issue, but this introduced this issue: https://github.com/openmrs/openmrs-distro-referenceapplication/pull/885#discussion_r1943002972
3) Meanwhile there was this commit e73c95124ce71518220780361a89f199de9fe578 which meant to fix the original bug, but ended up introducing more bugs.

vasharma05 commented:
Thanks @jayasanka-sack
